packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -447,6 +444,36 @@ export class Pipeline extends PipelineBase {
     return !owner || owner === 'AWS';
   }
 
+  private getOtherStackIfActionIsCrossAccount(action: IAction): Stack | undefined {
+    const pipelineStack = Stack.of(this);
+
+    if (action.actionProperties.resource) {
+      const resourceStack = Stack.of(action.actionProperties.resource);
+      // check if resource is from a different account
+      if (pipelineStack.account !== resourceStack.account) {
+        return resourceStack;
+      }
+    } else if (action.actionProperties.account &&
+        pipelineStack.account !== action.actionProperties.account) {
+      const targetAccount = action.actionProperties.account;
+      let targetAccountStack: Stack | undefined = this._crossAccountSupport[targetAccount];
+      if (!targetAccountStack) {
+        const app = this.requireApp();
+        targetAccountStack = new Stack(app, `cross-account-support-stack-${targetAccount}`, {
```

Comment:
No need to add the stack directly under the app. Add it inside the pipeline's scope

Reply:
Done.

Reply:
Scratch that. This does not work for some reason. The unit test in the `codepipeline-actions` package that tests the new behavior for the CFN actions fail when anchoring the new stack under the pipeline:

```
Error: Can only reference cross stacks in the same region and account. target = PipelineStack/Pipeline/cross-account-support-stack-123456789012/PipelineStackPipeline9DB740AF-Deploy-CFN-DeploymentRole/Resource
```

It seems to me like the cross-account behavior is not triggered at all when I add the new stack under the pipeline's scope (this is an error message from `CfnReference`).

Any ideas why might that be...? Could it be because `Stack.of(...)` does not work correctly if stacks are nested under constructs...?

Reply:
I think you are right. I think there is a bug in how substacks consume references. I will run some more tests tomorrow morning and hope to have some better understanding of this.

Reply:
This actually makes sense. The pipeline stack references the role in the other account and the automatic cross-stack references are triggered, but those only work for same-environment since they use exports/imports.

Digging further...

Reply:
Okay, haven't managed to figure out what the issue is. I still think there is a bug somewhere, but you'll have to dig further from your end. I've added a bunch of tests to verify some assumptions: https://github.com/aws/aws-cdk/pull/3373

Reply:
Ok. I've updated the PR to fix the out of order imports. Let's see if the build passes this time.